skfda/preprocessing/dim_reduction/feature_extraction/_ddg_transformer.py

```diff
@@ -0,0 +1,122 @@
+
+default_depth = ModifiedBandDepth()
+
+
+class DDGTransformer(BaseEstimator, TransformerMixin):
+    r"""Generalized depth-versus-depth (DD) transformer for functional data.
+
+    This transformer takes a list of k depths and performs the following map:
+
+    .. math::
+        \mathcal{X} &\rightarrow \mathbb{R}^G \\
+        x &\rightarrow \textbf{d} = (D_1^1(x), D_1^2(x),...,D_g^k(x))
+
+    Where :math:`D_i^j(x)` is the depth of the point :math:`x` with respect to
+    the data in the :math:`i`-th group using the :math:`j`-th depth of the
+    provided list.
+
+    Note that :math:`\mathcal{X}` is possibly multivariate, that is,
+    :math:`\mathcal{X} = \mathcal{X}_1 \times ... \times \mathcal{X}_p`.
+
+    Parameters:
+        depth_method:
+            The depth class or sequence of depths to use when calculating
+            the depth of a test sample in a class. See the documentation of
+            the depths module for a list of available depths. By default it
+            is ModifiedBandDepth.
+
+    Examples:
+        Firstly, we will import and split the Berkeley Growth Study dataset
+
+        >>> from skfda.datasets import fetch_growth
+        >>> from sklearn.model_selection import train_test_split
+        >>> dataset = fetch_growth()
+        >>> fd = dataset['data']
+        >>> y = dataset['target']
+        >>> X_train, X_test, y_train, y_test = train_test_split(
+        ...     fd, y, test_size=0.25, stratify=y, random_state=0)
+
+        >>> from skfda.preprocessing.dim_reduction.feature_extraction import \
+        ... DDGTransformer
+        >>> from sklearn.pipeline import make_pipeline
+        >>> from sklearn.neighbors import KNeighborsClassifier
+
+        We classify by first transforming our data using the defined map
+        and then using KNN
+
+        >>> pipe = make_pipeline(DDGTransformer(), KNeighborsClassifier())
+        >>> pipe.fit(X_train, y_train)
+        Pipeline(steps=[('ddgtransformer',
+                         DDGTransformer(depth_method=[ModifiedBandDepth()])),
+                        ('kneighborsclassifier', KNeighborsClassifier())])
+
+        We can predict the class of new samples
+
+        >>> pipe.predict(X_test)
+        array([1, 1, 1, 0, 1, 1, 1, 1, 0, 0, 0, 1,
+               1, 0, 0, 0, 0, 1, 1, 1, 1, 0, 1, 1])
+
+        Finally, we calculate the mean accuracy for the test data
+
+        >>> pipe.score(X_test, y_test)
+        0.875
+
+    References:
+        Cuesta-Albertos, J. A., Febrero-Bande, M. and Oviedo de la Fuente, M.
+        (2017). The DDG-classifier in the functional setting.
+        TEST, 26. 119-142.
+    """
+
+    def __init__(
+        self,
+        depth_method: Union[Depth, Sequence[Depth]] = default_depth,
+    ) -> None:
+        if isinstance(depth_method, Depth):
+            depth_method = [depth_method]
+        self.depth_method = depth_method
+
+    def fit(self, X: FDataGrid, y: ndarray) -> 'DDGTransformer':
```

Comment:
Use `from __future__ import annotations` instead of ugly strings as types.

Reply:
Thank you!